### PR TITLE
Telemetry for torch.compile()

### PR DESCRIPTION
**Pull request by popovaan**
### Details:
 - Send telemetry if openvino is used from torch.compile().

### Tickets:
 - CVS-134112

### CODE REVIEW

File: tools/ovc/openvino/tools/ovc/__init__.py [hunk 1]
```diff
@@ -18,3 +19,8 @@
 if is_optimum() and optimum_version is not None:
     telemetry = init_ovc_telemetry("Optimum Intel", optimum_version)
     telemetry.send_event("optimum", "import", "import_from_optimum,ov_version:{}".format(get_rt_version()))
+
+if is_torch() and 'torch' in sys.modules:
+    torch_version = importlib_metadata.version("torch")
+    telemetry = init_ovc_telemetry("PyTorch", torch_version)
+    telemetry.send_event("torch", "import", "Import from torch.compile")
```
Comment (mvafin) [inline tools/ovc/openvino/tools/ovc/__init__.py:26]: Do we want to include ov version like is done with optimum?

Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/__init__.py]: Done.

**Review by mvafin — APPROVED** [review]

File: tools/ovc/openvino/tools/ovc/__init__.py [hunk 2]
```diff
@@ -18,3 +19,9 @@
 if is_optimum() and optimum_version is not None:
     telemetry = init_ovc_telemetry("Optimum Intel", optimum_version)
     telemetry.send_event("optimum", "import", "import_from_optimum,ov_version:{}".format(get_rt_version()))
+
+if is_torch() and 'torch' in sys.modules:
+    from openvino.runtime import get_version as get_rt_version  # pylint: disable=no-name-in-module,import-error
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/__init__.py:24]:
```suggestion
    from openvino import get_version as get_rt_version  # pylint: disable=no-name-in-module,import-error
```

File: tools/ovc/openvino/tools/ovc/__init__.py [hunk 3]
```diff
@@ -18,3 +19,9 @@
 if is_optimum() and optimum_version is not None:
     telemetry = init_ovc_telemetry("Optimum Intel", optimum_version)
     telemetry.send_event("optimum", "import", "import_from_optimum,ov_version:{}".format(get_rt_version()))
+
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/__init__.py:22]: I think it is better to rename `is_torch` to `is_torch_compile`. How do you think?

File: tools/ovc/openvino/tools/ovc/__init__.py [hunk 4]
```diff
@@ -18,3 +19,9 @@
...
+if is_torch() and 'torch' in sys.modules:
+    from openvino.runtime import get_version as get_rt_version  # pylint: disable=no-name-in-module,import-error
+    torch_version = importlib_metadata.version("torch")
+    telemetry = init_ovc_telemetry("PyTorch", torch_version)
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/__init__.py:26]:
```suggestion
    telemetry = init_ovc_telemetry("torch.compile", torch_version)
```

File: tools/ovc/openvino/tools/ovc/__init__.py [hunk 5]
```diff
@@ -18,3 +19,9 @@
...
+    telemetry.send_event("torch", "import", "Import from torch.compile(), ov_version:{}".format(get_rt_version()))
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/__init__.py:27]:
```suggestion
    telemetry.send_event("torch.compile", "import", "Import from torch.compile(), ov_version:{}".format(get_rt_version()))
```

File: tools/ovc/openvino/tools/ovc/__init__.py (hunk 3 above)
Comment (rkazants) [reply tools/ovc/openvino/tools/ovc/__init__.py:22]: it is `nit` and not possible to do right now.

File: tools/ovc/openvino/tools/ovc/__init__.py (hunk 2 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/__init__.py]: Removed this import, as it is redundant and applied this comment in the import above.

File: tools/ovc/openvino/tools/ovc/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/__init__.py]: Done.

File: tools/ovc/openvino/tools/ovc/__init__.py (hunk 4 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/__init__.py]: Done.

File: tools/ovc/openvino/tools/ovc/__init__.py (hunk 5 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/__init__.py]: Done.

**Review by rkazants — APPROVED** [review]

**Pull request merged**